integrationTests/relayers/slowTests/framework/multiversxHandler.go

```diff
@@ -800,21 +803,41 @@ func (handler *MultiversxHandler) setInitialSupply(ctx context.Context, params I
 		require.True(handler, okConvert)
 
 		if params.IsMintBurnOnMvX {
-			hash, txResult := handler.ChainSimulator.ScCall(
-				ctx,
-				handler.OwnerKeys.MvxSk,
-				handler.MultisigAddress,
-				zeroStringValue,
-				setCallsGasLimit,
-				initSupplyMintBurnEsdtSafe,
-				[]string{
-					hex.EncodeToString([]byte(tkData.MvxChainSpecificToken)),
-					hex.EncodeToString(initialSupply.Bytes()),
-					hex.EncodeToString([]byte{0}),
-				},
-			)
-			log.Info("initial supply tx executed", "hash", hash, "status", txResult.Status,
-				"initial mint", params.InitialSupplyValue, "initial burned", "0")
+			if params.IsNativeOnMvX {
+				burnAmount := initialSupply
+				hash, txResult := handler.ChainSimulator.ScCall(
+					ctx,
+					handler.OwnerKeys.MvxSk,
+					handler.MultisigAddress,
+					zeroStringValue,
+					setCallsGasLimit,
+					initSupplyMintBurnEsdtSafe,
+					[]string{
+						hex.EncodeToString([]byte(tkData.MvxChainSpecificToken)),
+						hex.EncodeToString(zeroValueBigInt.Bytes()),
+						hex.EncodeToString(burnAmount.Bytes()),
+					},
+				)
+				log.Info("initial supply tx executed", "hash", hash, "status", txResult.Status,
+					"initial mint", "0", "initial burned", params.InitialSupplyValue)
+			} else {
+				mintAmount := initialSupply
+				hash, txResult := handler.ChainSimulator.ScCall(
+					ctx,
+					handler.OwnerKeys.MvxSk,
+					handler.MultisigAddress,
+					zeroStringValue,
+					setCallsGasLimit,
+					initSupplyMintBurnEsdtSafe,
+					[]string{
+						hex.EncodeToString([]byte(tkData.MvxChainSpecificToken)),
+						hex.EncodeToString(mintAmount.Bytes()),
+						hex.EncodeToString(zeroValueBigInt.Bytes()),
+					},
+				)
+				log.Info("initial supply tx executed", "hash", hash, "status", txResult.Status,
+					"initial mint", params.InitialSupplyValue, "initial burned", "0")
+			}
```

Comment:
same here, you can call ScCall only once, just set the proper variables inside ifs